File: languages/php/worker.php
<?php
const KEY=0x83; while(($line=fgets(STDIN))!==false){$line=rtrim($line,"\r\n");if($line==='PING'){echo "PONG\n";flush();continue;}if($line==='QUIT')break;if(preg_match('/^[ED] ([0-9a-fA-F]*)$/',$line,$m)){ $h=$m[1];$o='';for($i=0;$i<strlen($h);$i+=2)$o.=sprintf('%02x',hexdec(substr($h,$i,2))^KEY);echo "$o\n";flush();}else{echo "ERR\n";flush();}}
?>

File: languages/python/worker.py
#!/usr/bin/env python3
import sys
KEY=0x59
for line in sys.stdin:
    line=line.rstrip("\r\n")
    if line=="PING": print("PONG",flush=True); continue
    if line=="QUIT": break
    if len(line)>=2 and line[0] in "ED" and line[1]==" ":
        try: print(bytes(b^KEY for b in bytes.fromhex(line[2:])).hex(),flush=True)
        except ValueError: print("ERR",flush=True)
    else: print("ERR",flush=True)
